Move internal feature enablement flags to Open Source

brentfulgham opened the pull request:
#### b93d3a485f777e9d81ca56dfb011aeeff66ec8e9
<pre>
Move internal feature enablement flags to Open Source
<a href="https://bugs.webkit.org/show_bug.cgi?id=261477">https://bugs.webkit.org/show_bug.cgi?id=261477</a>
&lt;rdar://115387892&gt;

Reviewed by Tim Horton.

Now that the public betas of our iOS 17 and macOS 14 releases have been shipping
for a few months, we should move all relevant flags into Open Source to reduce
maintenance burden.

* Source/WTF/wtf/PlatformEnable.h:
* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WTF/wtf/PlatformHave.h:
* Source/WTF/wtf/PlatformUse.h:
* Source/WebKit/NetworkProcess/cocoa/NetworkSessionCocoa.mm
(WebKit::activateSessionCleanup): Correct PLATFORM macro uses
* Source/WebKit/Platform/spi/ios/UIKitSPI.h: UIDatePickerStyle is now API on all build targets,
so this declaration is unneeded.
* Source/WebKit/Platform/spi/watchos/PepperUICoreSPI.h: Add missing forward declarations.
* Source/WebKit/UIProcess/ios/forms/WKSelectMenuListViewController.mm: Correct include
for watchOS builds.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/PepperUICore.framework/PepperUICore.tbd: Add missing class declaration.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/PrivateFrameworks/PepperUICore.framework/PepperUICore.tbd: Ditto.

Canonical link: <a href="https://commits.webkit.org/268103@main">https://commits.webkit.org/268103@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/20167984e8944acee2cfd8ddf95585cd90a0934a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/18681 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/19022 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/19625 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/20544 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/17487 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/18877 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/22331 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/19162 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/19327 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/18905 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/19061 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/16269 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/21421 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/16284 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/17029 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/23478 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/16210 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/17309 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/17201 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/21371 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/18010 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/17797 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/15109 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/22050 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/16854 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/5380 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4445 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/21221 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/23295 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/17635 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/5235 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

hortont424 commented on Source/WTF/wtf/PlatformEnableCocoa.h:
```diff
@@ -903,3 +916,77 @@
+
+#if !defined(ENABLE_NETWORK_ISSUE_REPORTING) \
+    && ((PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 140000) \
+    || ((PLATFORM(IOS) || PLATFORM(MACCATALYST)) && __IPHONE_OS_VERSION_MIN_REQUIRED >= 170000) \
```
Soooo close to just being IOS_FAMILY

brentfulgham replied:
Yeah, the pesky builders have to be kept happy. We should do a revision once the bots have moved to fall release.

hortont424 commented on Source/WTF/wtf/PlatformHave.h:
```diff
@@ -1269,6 +1277,30 @@
+#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 130000) \
+    || ((PLATFORM(IOS) || PLATFORM(MACCATALYST)) && __IPHONE_OS_VERSION_MIN_REQUIRED >= 160000) \
+    || PLATFORM(VISION) \
+    || (PLATFORM(WATCHOS) && __WATCH_OS_VERSION_MIN_REQUIRED >= 90000) \
```
I doubt we need these watch and TV version checks

hortont424 commented on Source/WTF/wtf/PlatformHave.h:
```diff
@@ -1512,3 +1559,27 @@
+
+#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 140000) \
+    || (PLATFORM(MACCATALYST) && __IPHONE_OS_VERSION_MIN_REQUIRED >= 170000) \
+    || PLATFORM(IOS) \
```
This looks like IOS_FAMILY to me?

brentfulgham replied:
I ended up needing to keep this separate for now, since the different IOS_FAMILY things are gated on different versions, and failing to do so broke the Open Source builders.

brentfulgham committed:
Move internal feature enablement flags to Open Source
https://bugs.webkit.org/show_bug.cgi?id=261477
<rdar://115387892>

Reviewed by Tim Horton.

Now that the public betas of our iOS 17 and macOS 14 releases have been shipping
for a few months, we should move all relevant flags into Open Source to reduce
maintenance burden.

* Source/WTF/wtf/PlatformEnable.h:
* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WTF/wtf/PlatformHave.h:
* Source/WTF/wtf/PlatformUse.h:
* Source/WebKit/NetworkProcess/cocoa/NetworkSessionCocoa.mm
(WebKit::activateSessionCleanup): Correct PLATFORM macro uses
* Source/WebKit/Platform/spi/ios/UIKitSPI.h: UIDatePickerStyle is now API on all build targets,
so this declaration is unneeded.
* Source/WebKit/Platform/spi/watchos/PepperUICoreSPI.h: Add missing forward declarations.
* Source/WebKit/UIProcess/ios/forms/WKSelectMenuListViewController.mm: Correct include
for watchOS builds.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/PepperUICore.framework/PepperUICore.tbd: Add missing class declaration.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/PrivateFrameworks/PepperUICore.framework/PepperUICore.tbd: Ditto.

Canonical link: https://commits.webkit.org/268103@main
